docs/how_to/aggregator.md

```diff
@@ -0,0 +1,154 @@
+# Setting up m3aggregator
+
+## Introduction
+
+m3aggregator is used to cluster stateful downsampling and rollup of metrics before they are store in M3DB. The m3coordinator also performs this role but is not cluster aware. This means metrics will not get aggregated properly if you send metrics in round round fashion to multiple m3coordinators for the same metrics ingestion source (e.g. Prometheus server).
+
+Metrics sent to m3aggregator are correctly routed to the instance resposible for aggregating each metric and there is multiple m3aggregator replicas to ensure no single point of failure during aggregation.
+
+## Configuration
+
+Before setting up m3aggregator, make sure that you have at least [one M3DB node running](single_node.md) and a dedicated m3coordinator setup.
+
+We highly recommend running with at least a replication factor 2 for a m3aggregator deployment.
+
+### Topology
+
+#### Initializing aggregator topology 
+You can setup a m3aggregator topology by issuing a request to your coordinator (be sure to use your own hostnames, number of shards and replication factor):
+```bash
+curl -vvvsSf -H "Cluster-Environment-Name: namespace/m3db-cluster-name" -X POST http://m3coordinator01:7201/api/v1/services/m3aggregator/placement/init -d '{
+    "num_shards": 64,
+    "replication_factor": 2,
+    "instances": [
+        {
+            "id": "m3aggregator01:6000",
+            "isolation_group": "availability-zone-a",
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator01:6000",
+            "hostname": "m3aggregator01",
+            "port": 6000
+        },
+        {
+            "id": "m3aggregator02:6000",
+            "isolation_group": "availability-zone-b",
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator02:6000",
+            "hostname": "m3aggregator02",
+            "port": 6000
+        }
+    ]
+}'
```

Comment:
(off-topic) I know that the goal of this PR is to only document the process, however, I was wondering if the components can self-register with each other. Something like automated service-discovery process. WDYT?

Reply:
That would be ideal yes, at some point we could look at Kubernetes.